### PR TITLE
refactor(ui): use `import type` syntax where possible

### PR DESCRIPTION
**Pull request by agilgur5**
Unblocks #12516
Follow-up to #12163 & #12290, which unblocked this
Follow-up to https://github.com/argoproj/argo-workflows/pull/12150#discussion_r1382510455 
Same changes can be made for #12158 and the `argo-ui` warnings introduced in #12061

### Motivation

- now that we've migrated from archived `tslint` to `eslint` and updated `prettier` from v1 to v3, we can _finally_ use this [newer TS syntax from TS 3.8](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export)
  - this fixes a few warnings in the build as well as makes things more explicit and efficient
    - **note**: there are more warnings to fix around `export type`, but those are all from `argo-ui`'s source code and so can't be fixed from within this repo

### Modifications

- use `import type` syntax in a few files where it is only types that are imported

### Verification

1. `yarn build` works fine, with a few less warnings now
2. `make start UI=true` and tested UI, esp. `monaco-editor`, all still works fine

### CODE REVIEW

**Review by toyamagu-2021 — APPROVED** [review]
Great! LGTM

**Review by terrytangyuan — COMMENTED** [review]
Why is this a fix?

**Comment by agilgur5** [comment]
> Why is this a fix?

because it fixes existing warnings. those warnings also become errors in #12516

**Review by isubasinghe — APPROVED** [review]
LGTM

**Comment by agilgur5** [comment]
> could potentially word it as a `build` commit, although it doesn't directly touch the build, the changes are to the source code.

I re-worded it as a `refactor` so that it's not cherry-picked into patches since it relies on other build changes (#12163 and #12290)

**Comment by juliev0** [comment]
How would you feel about being assignee on this @terrytangyuan ? Given that it's a "prioritized-review", and has been for over a week, it needs an Assignee. I don't think I have the right background for this. But I see it already has 2 approvals. It also looks like this other prioritized-review [PR](https://github.com/argoproj/argo-workflows/pull/12516) is ready but blocked on this one.

**Comment by terrytangyuan** [comment]
I have a concern on these small refactor PRs. Started discussion on Slack https://cloud-native.slack.com/archives/C0510EUH90V/p1706279158785129

**Comment by juliev0** [comment]
@isubasinghe Are you comfortable merging this since you've given approval?

**Comment by isubasinghe** [comment]
@juliev0 I am comfortable merging this, but I don't have write access yet I think?

**Comment by juliev0** [comment]
Got it! Okay, should be soon.On Feb 2, 2024, at 6:53 PM, Isitha Subasinghe ***@***.***> wrote:﻿
@juliev0 I am comfortable merging this, but I don't have write access yet I think?

**Review by terrytangyuan — APPROVED** [review]

**Comment by agilgur5** [comment]
> > @juliev0 I am comfortable merging this, but I don't have write access yet I think?

> Got it! Okay, should be soon.

Crenshaw said he'd get to membership etc updates on Monday, for reference

**Comment by terrytangyuan** [comment]
I’ll add once it’s publicly announced

**Pull request merged**
